mirgecom/eos.py

```diff
@@ -172,6 +214,21 @@ def get_production_rates(self, cv: ConservedVars):
     def get_species_source_terms(self, cv: ConservedVars):
         r"""Get the species mass source terms to be used on the RHS for chemistry."""
 
+    def dependent_vars(self, cv: ConservedVars,
+                       temperature_seed: DOFArray = None) -> MixtureDependentVars:
+        """Get an agglomerated array of the dependent variables.
+
+        Certain implementations of :class:`GasEOS` (e.g. :class:`MixtureEOS`)
+        may raise :exc:`TemperatureSeedError` if *temperature_seed* is not
+        given.
+        """
+        return MixtureDependentVars(
+            temperature=self.temperature(cv, temperature_seed),
+            pressure=self.pressure(cv),
```

Comment:
`pressure` seems to internally access a memoized temperature. I think we should get rid of that `memoize` and instead make the functional dependency of pressure on temperature explicit.

Reply:
Agreed: Here's an associated issue (#574 ). I have fixed this, and tested it.   Seems to work ok.   The use of `gas_model` and `make_fluid_state` vastly reduces the number of calls to the EOS functions by itself.   This will be fixed up in an independent, upcoming PR.

Reply:
I apologize for my flipflop on this.  Following the advice of @thomasgibson , I am going to make this state-handling change as close to the end-state as possible to avoid multiple follow-on's that make significant changes.   As such, this PR will *de-memoize*.  (6e02d94).   

It's down to the production drivers now.   It seems one or more of them are having their way with the EOS and we need to put a stop to it.